framework/include/problems/DisplacedProblem.h

```diff
@@ -403,6 +404,54 @@ class DisplacedProblem : public SubProblem
 
   GeometricSearchData _geometric_search_data;
 
+  class UpdateDisplacedMeshThread : public ThreadedNodeLoop<NodeRange, NodeRange::const_iterator>
+  {
+  public:
+    UpdateDisplacedMeshThread(FEProblemBase & fe_problem, DisplacedProblem & displaced_problem);
+
+    UpdateDisplacedMeshThread(UpdateDisplacedMeshThread & x, Threads::split split);
+
+    virtual void onNode(NodeRange::const_iterator & nd) override;
+
+    void join(const UpdateDisplacedMeshThread & y)
+    {
+      if (y._has_displacement)
+        _has_displacement = true;
+    }
+
+    /**
+     * Whether the displaced mesh is modified by the latest call to operator()
+     */
+    bool hasDisplacement()
+    {
+      _ref_mesh.comm().max(_has_displacement);
```

Comment:
let's assert not in libmesh::in_threads() (or something similar)
since it bit you before

Reply:
Added the assertion here. Maybe we should add this in libMesh::Communicator functions.